pkg/skaffold/kubernetes/portforward/entry_manager.go

```diff
@@ -31,28 +31,124 @@ import (
 var (
 	// For testing
 	forwardingTimeoutTime = time.Minute
+	portForwardEvent      = func(entry *portForwardEntry) {
+		// TODO priyawadhwa@, change event API to accept ports of type int
+		event.PortForwarded(
+			int32(entry.localPort),
+			int32(entry.resource.Port),
+			entry.podName,
+			entry.containerName,
+			entry.resource.Namespace,
+			entry.portName,
+			string(entry.resource.Type),
+			entry.resource.Name)
+	}
 )
 
+type forwardedPorts struct {
+	ports map[int]struct{}
+	lock  *sync.Mutex
+}
+
+func (f forwardedPorts) Store(key, value interface{}) {
+	f.lock.Lock()
+	defer f.lock.Unlock()
+	val, ok := key.(int)
+	if !ok {
+		panic("only store keys of type int in forwardedPorts")
+	}
+	f.ports[val] = struct{}{}
```

Comment:
is this a bug? shouldn't we store value instead of empty struct?

Reply:
whoops, val should be renamed to `k`, thanks for catching this!